### PR TITLE
test(client): custom output with yarn

### LINKED ISSUE #17685: .prisma/client: Name can't start with a dot
### Bug description

When prima client is generated with custom output path, it throws error `.prisma/client: Name can't start with a dot`
Screenshot 2023-02-02 at 12 29 24 AM

### Expected behavior

I want to run yarn normally after manually generating the client.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
  output   = "../src/generated/client"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v16.13.2

### Prisma Version

```
prisma                  : 4.9.0
@prisma/client          : 4.9.0
```

### PR DESCRIPTION
**Pull request by millsp**
closes https://github.com/prisma/team-orm/issues/866
closes https://github.com/prisma/prisma/issues/17685

### CODE REVIEW

File: packages/client/src/generation/generateClient.ts [hunk 1]
```diff
@@ -133,6 +133,7 @@ export async function buildClient({
     types: 'index.d.ts',
     browser: 'index-browser.js',
     exports: clientPackageExports,
+    version: clientVersion,
```
Comment (millsp) [inline packages/client/src/generation/generateClient.ts:136]: Found that this was missing when writing the test.

Comment (millsp) [reply packages/client/src/generation/generateClient.ts:136]: I had to put the PR in draft again. Somehow changing either `version` or `private` causes our functional type tests to fail. I don't understand how they influence each other.

File: packages/client/tests/e2e/custom-output-yarn/readme.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Readme
+
+This tests ensures that custom output clients are working as expected with Yarn.
```
Comment (Jolg42) [inline packages/client/tests/e2e/custom-output-yarn/readme.md:3]:
```suggestion
This tests ensures that custom output clients are working as expected with Yarn 1.
```
I'm not sure how this goes with other versions

File: packages/client/tests/e2e/custom-output-yarn/readme.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+A package is deemed not to be valid by yarn if:
```
Comment (Jolg42) [inline packages/client/tests/e2e/custom-output-yarn/readme.md:5]:
```suggestion
A `package.json` is invalid for `yarn` if:
```

File: packages/client/tests/e2e/custom-output-yarn/readme.md (hunk 2 above)
Comment (SevInf) [reply packages/client/tests/e2e/custom-output-yarn/readme.md:3]: Same with Berry

File: packages/client/src/generation/generateClient.ts (hunk 1 above)
Comment (SevInf) [reply packages/client/src/generation/generateClient.ts:137]: So, i figured that out and fixed. It's quite a wild thing: If we have multiple npm packages with the same name and version in the tree and they all contain the same namespace declaration, TS will merge those together into single namespace. Not having a `version` in generated `package.json` prevented that. Now, when we added the version, all `Prisma` namespace declaration from all of the tests got merged together, sometimes overriding each other's properties, sometimes producing intersections that result in `never`` and so on. 

I've fixed that in my commit here by appending sha256 of the schema file to a generated package name. We might also consider making generated package name configurable.

**Review by Jolg42 — APPROVED** [review]

**Pull request merged**